solutions/devsprint-firas-safa-1/GitHubApp/Data/Network/NetworkService.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  NetworkEngine.swift
+//  GitHubApp
+//
+//  Created by Razee Hussein-Jamal on 09/02/2022.
+//
+
+import Foundation
+
+
+protocol NetworkRequestProtocol {
+    func makeGetRequest<ResponseType: Decodable>(with request: URLRequest, using session: URLSession, completion: @escaping (GithubResult<ResponseType>) -> Void)
```

Comment:
Fix naming

Reply:
Move the response type to the protocol you are going to create

Reply:
@TheInkedEngineer do you recommend to change the closure based code into async await?